src/Server/middleware/linkHeadersMiddleware.ts

```diff
@@ -0,0 +1,28 @@
+import type { NextFunction } from "express"
+import type { ArtsyRequest, ArtsyResponse } from "./artsyExpress"
+
+import { CDN_URL, GEMINI_CLOUDFRONT_URL, WEBFONT_URL } from "Server/config"
+
+/**
+ * Link headers allow 103: Early Hints to be sent to the client (by Cloudflare).
+ * These should correspond to preconnect and preload items from html.ejs.
+ * See: https://developers.cloudflare.com/cache/advanced-configuration/early-hints/
+ */
+export function linkHeadersMiddleware(
+  req: ArtsyRequest,
+  res: ArtsyResponse,
+  next: NextFunction
+) {
+  if (!res.headersSent) {
+    res.header("Link", [
+      `<${CDN_URL}>; rel=preconnect; crossorigin`,
+      `<${GEMINI_CLOUDFRONT_URL}>; rel=preconnect; crossorigin`,
+      `<${WEBFONT_URL}>; rel=preconnect; crossorigin`,
+      `<${WEBFONT_URL}/all-webfonts.css>; rel=preload; as=style`,
+      `<${WEBFONT_URL}/ll-unica77_regular.woff2>; rel=preload; as=font; crossorigin`,
+      `<${WEBFONT_URL}/ll-unica77_medium.woff2>; rel=preload; as=font; crossorigin`,
+      `<${WEBFONT_URL}/ll-unica77_italic.woff2>; rel=preload; as=font; crossorigin`,
+    ] as any)
```

Comment:
The `header` or `set` methods _do_ in fact accept an array. I attempted to update the types:

```bash
yarn add -D @types/express@4.17.21
```

...but that encountered many more errors of the form:

```
The inferred type of 'adminServerRoutes' cannot be named without a reference to 'express/node_modules/@types/express-serve-static-core'. This is likely not portable. A type annotation is necessary.
```

Reply:
Lemme try locally real quick; this could likely be a quick fix (though any is totally fine here). One moment...

Reply:
Pushed two small commits!